### PR TITLE
Improve errors: Data and mesh config

### PR DESCRIPTION
**Pull request by MakisH**
```diff
- Data "MyData" cannot be created twice for mesh "MyMesh"!
+ Data "MyData" cannot be created twice for mesh "MyMesh". Please rename or remove one of the use-data tags with name "MyData".
```
```diff
- Data "MyData" uses non-unique name!
+ Data "MyData" has already been defined. Please rename or remove one of the data tags with name="MyData".
```
```diff
- Data "MyData" is not available in data configuration!
+ Data "MyData" is not defined. Please define a data tag with name="MyData".
```
```diff
- Unknown data type!
+ Unknown data type "typeName". Known data types: "scalar", "vector".
```

```diff
- Data with name "MyData" is not available during configuration of mesh "MyMesh"
+ Data with name "MyData" used by mesh "MyMesh" is not defined. Please define a data tag with name="MyData"
```

Related to #698.

### CODE REVIEW

File: src/mesh/config/MeshConfiguration.cpp [hunk 1]
```diff
@@ -109,7 +110,7 @@ void MeshConfiguration::addMesh(
         break;
       }
     }
-    PRECICE_CHECK(found, "Data " << dataNewMesh->getName() << " is not available in data configuration!");
+    PRECICE_CHECK(found, "Data " << dataNewMesh->getName() << " is not defined. Please define a data tag with name=\""<< dataNewMesh->getName() << "\".");
```
Comment (KyleDavisSA) [inline src/mesh/config/MeshConfiguration.cpp:113]: Seems like this error won't be run into, but good to keep it? I always hit the error above first.

Comment (MakisH) [reply src/mesh/config/MeshConfiguration.cpp:113]: I also don't understand why this check is here, but I also don't see any problem with keeping it. Let's keep it then?

Comment (MakisH) [reply src/mesh/config/MeshConfiguration.cpp:118]: @KyleDavisSA @uekerman shall we keep this then?

**Comment by MakisH** [comment]
I rebased on the latest develop. The files still need formatting, but maybe that would be confusing, as this would add also irrelevant changes in the diff.

File: src/mesh/config/MeshConfiguration.cpp (hunk 1 above)
Comment (uekerman) [reply src/mesh/config/MeshConfiguration.cpp:118]: Seems like this function is only called in tests. We could convert it to an assertion.

Comment (MakisH) [reply src/mesh/config/MeshConfiguration.cpp]: I converted this to an assertion, thanks!

**Pull request merged**